pkg/policy/mapstate_test.go

```diff
@@ -1571,7 +1535,7 @@ func TestMapState_AccumulateMapChanges(t *testing.T) {
 		state: testMapState(map[Key]mapStateEntry{
 			egressKey(43, 0, 0, 0):  allowEntry(0, csFoo).WithAuthType(AuthTypeAlwaysFail),
 			egressKey(0, 6, 0, 0):   allowEntry(0, csFoo).WithAuthType(AuthTypeSpire),
-			egressKey(43, 6, 80, 0): allowEntry(1, csBar).WithDefaultAuthType(AuthTypeSpire),
+			egressKey(43, 6, 80, 0): allowEntry(1, csBar).WithDefaultAuthType(AuthTypeAlwaysFail),
```

Comment:
Is this a bug in the test?

Reply:
No, this is due to the auth type now being derived only from the key with the same ID. Now the datapath is responsible for overriding the explicit spire auth type from the L4-only key when both L4-only and L3-keys match.